modules/index/src/integrationTest/java/org/apache/ignite/internal/index/ItBuildIndexTest.java

```diff
@@ -133,23 +137,115 @@ void testDropIndexDuringTransaction() throws Exception {
 
         checkIndexBuild(partitions, replicas, INDEX_NAME);
 
+        CompletableFuture<Void> indexRemovedFuture = indexRemovedFuture();
+
         IgniteImpl node = CLUSTER.aliveNode();
 
         // Start a transaction. We expect that the index will not be removed until this transaction completes.
-        Transaction tx = node.transactions().begin(new TransactionOptions().readOnly(false));
+        node.transactions().runInTransaction(tx -> {
+            dropIndex(INDEX_NAME);
+
+            CatalogIndexDescriptor indexDescriptor = node.catalogManager().index(INDEX_NAME, node.clock().nowLong());
+
+            assertThat(indexDescriptor, is(notNullValue()));
+            assertThat(indexDescriptor.status(), is(CatalogIndexStatus.STOPPING));
```

Comment:
Maybe I just didn’t have time to change the status, maybe add some other check or wait a little? what do you think?

Reply:
Fixed